src/Fantomas/CodePrinter.fs

```diff
@@ -4103,6 +4103,7 @@ and genClause astContext hasBar (Clause (p, e, eo)) =
         +> leaveNodeTokenByName arrowRange RARROW
 
     genTriviaBeforeClausePipe p.Range
+    +> genTriviaMainNodesBeforeClausePipe p.Range
```

Comment:
You introduce a very specific helper here. I'd prefer that an existing helper is used to print the trivia.
`|> genTriviaFor SynMatchClause_Clause p.Range`.

However, this does not solve your indentation problem.
I would add an extra check if you have TriviaContentBefore the `SynMatchClause`.

https://github.com/fsprojects/fantomas/blob/ca42f0d7cf05144c5838bba725378e2a856211bb/src/Fantomas/CodePrinter.fs#L2040

Something like

```fsharp
            let hasCommentBeforeClause = true // TODO: add actual check in TriviaHelpers
            let genClause (astContext: ASTContext) (b: bool) (c: SynMatchClause) =
                ifElse
                    hasCommentBeforeClause
                    (indent +> genClause astContext b c +> unindent)
                    (genClause astContext b c)
```

And if that is the case wrap the genClause between indents.